### PR TITLE
chore: update to testcontainers-go v0.33.0

### LINKED ISSUE #6: Code is not compatible with testcontainers v0.31.0 and later
In file `dynamodb.go` we find these lines:

```go
// WithSharedDB allows container reuse between successive runs. Data will be persisted
func WithSharedDB() testcontainers.CustomizeRequestOption {

	return func(req *testcontainers.GenericContainerRequest) {
		if len(req.Cmd) > 0 {
			req.Cmd = append(req.Cmd, "-sharedDb")
		} else {
			req.Cmd = append(req.Cmd, "-jar", "DynamoDBLocal.jar", "-sharedDb")
		}
		req.Name = containerName
		req.Reuse = true
	}
}
```

The problem is that  `testcontainers.CustomizeRequestOption` is now a `func(*testcontainers.GenericContainerRequest) error` but the code is returning a `func (*testcontainers.GenericContainerRequest)` instead.

That breaking change was introduced in minor release v0.31.0 https://github.com/testcontainers/testcontainers-go/releases/tag/v0.31.0

### PR DESCRIPTION
**Pull request by mdelapenya**
@abhirockzz I think this PR is valid and needed by the community using Testcontainers in the most recent versions.

Could you please take a look and eventually merge it?

Thanks!

## Related issues
- Closes #5, because it introduces too many changes (options, different repo, etc)
- Caused by https://github.com/testcontainers/testcontainers-go/pull/2267
- Fixes #6 
- Supersedes and Closes #3

### CODE REVIEW

**Review by rafet-ongun — APPROVED** [review]

File: dynamodb_test.go
Comment (fgmarand) [inline dynamodb_test.go]: Maybe refactor the valid image+tag to a constant at the top of the file, or even in the non-test code, to make it easier to update in just one place, since it is used so many times ?

File: dynamodb_test.go
Comment (fgmarand) [inline dynamodb_test.go]: ...and refactor the invalid version next to the valid one, to make it more obvious.

Comment (mdelapenya) [reply dynamodb_test.go]: SGTM, thanks!

File: dynamodb_test.go
Comment (mdelapenya) [reply dynamodb_test.go]: SGTM, thanks!

**Review by fgmarand — APPROVED** [review]

**Comment by mdelapenya** [comment]
Added a commit to bump to the latest release, which we released today.

**Comment by mdelapenya** [comment]
@abhirockzz is there anything we can do for this one? Once merged, a release would be convenient too 🙏

**Comment by mdelapenya** [comment]
Friendly ping @abhirockzz 🙏

**Comment by mdelapenya** [comment]
I'm not sure about the activity in this repo, so I'd ask if anybody here would like to fork it and be the community maintainer. Else, we can adopt it and move it to the main testcontainers-go repo. Wdyt?

**Comment by fgmarand** [comment]
The second possibility seems to be more attractive for timeliness of updates.

**Comment by mdelapenya** [comment]
I submitted the module to the upstream project: https://github.com/testcontainers/testcontainers-go/pull/2799

**Comment by mdelapenya** [comment]
We merged the PR, and it will available in the next release of tc-go. We are changing the link in the modules catalog: https://testcontainers.com/modules/dynamodb/

Closing.

**Pull request closed without merging**
